dashboard/src/lib/hooks/useAppValidation.ts

```diff
@@ -70,12 +70,21 @@ export const useAppValidation = ({
       if (!deploymentTargetID) {
         throw new Error("No deployment target selected");
       }
+      const variables: Record<string, string> = {};
 
+      if (data.app.env) {
+        data.app.env?.forEach(item => {
+          if (item && !item.deleted) { // Ensure item exists and is not deleted
+
+            variables[item.key] = item.value;
+
+          }
+        });
+      }
```

Comment:
Not sure if you ran into a case where app.env was null but I think its of type array, so we might be able to get away without the check. Would also suggest the following:

```suggestion
      const variables = data.app.env
        .filter((e) => !e.deleted)
        .reduce((acc: Record<string, string>, curr) => {
          acc[curr.key] = curr.value;
          return acc;
        }, {});
```